sdk/storage/azure-storage-blob-nio/src/main/java/com/azure/storage/blob/nio/AzureFileSystemProvider.java

```diff
@@ -3,9 +3,299 @@
+ *     <li>{@code AzureStorageSasToken:}{@link String}</li>
+ *     <li>{@code AzureStorageHttpLogDetailLevel:}{@link com.azure.core.http.policy.HttpLogDetailLevel}</li>
+ *     <li>{@code AzureStorageMaxTries:}{@link Integer}</li>
+ *     <li>{@code AzureStorageTryTimeout:}{@link Integer}</li>
+ *     <li>{@code AzureStorageRetryDelayInMs:}{@link Long}</li>
+ *     <li>{@code AzureStorageMaxRetryDelayInMs:}{@link Long}</li>
+ *     <li>{@code AzureStorageRetryPolicyType:}{@link com.azure.storage.common.policy.RetryPolicyType}</li>
+ *     <li>{@code AzureStorageSecondaryHost:}{@link String}</li>
+ *     <li>{@code AzureStorageSecondaryHost:}{@link Integer}</li>
+ *     <li>{@code AzureStorageBlockSize:}{@link Integer}</li>
+ *     <li>{@code AzureStorageDownloadResumeRetries:}{@link Integer}</li>
+ *     <li>{@code AzureStorageUseHttps:}{@link Boolean}</li>
+ *     <li>{@code AzureStorageFileStores:}{@link Iterable<String>}</li>
+ * </ul>
+ * <p>
+ * Either an account key or a sas token must be specified. If both are provided, the account key will be preferred. If
+ * a sas token is specified, the customer must take care that it has appropriate permissions to perform the actions
+ * demanded of the file system in a given workflow, including the initial connection check specified above. Furthermore,
+ * it must have an expiry time that lasts at least until the file system is closed as there is no token refresh offered
+ * at this time. The same token will be applied to all containers.
+ * <p>
+ * An iterable of file stores must also be provided; each entry should simply be the name of a container. All other
+ * values listed are used to configure the underlying {@link com.azure.storage.blob.BlobServiceClient}. Please refer to
+ * that type for more information on these values.
+ *
+ * @see FileSystemProvider
  */
-public class AzureFileSystemProvider  {
+public final class AzureFileSystemProvider extends FileSystemProvider {
+    // TODO: Add logger
+    private static final String ACCOUNT_QUERY_KEY = "account";
+
+    private ConcurrentMap<String, FileSystem> openFileSystems;
+
+    // Specs require a public zero argument constructor.
+
+    /**
+     * Creates an AzureFileSystemProvider.
+     */
+    public AzureFileSystemProvider() {
+        this.openFileSystems = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Returns {@code "azb".}
+     */
+    @Override
+    public String getScheme() {
+        return "azb";
+    }
+
+    /**
+     * The format of a {@code URI} identifying a file system is {@code "azb://?account=&lt;accountName&gt;"}.
+     * <p>
+     * Once closed, a file system with the same identifier may be reopened.
+     * {@inheritDoc}
+     */
+    @Override
+    public FileSystem newFileSystem(URI uri, Map<String, ?> config) throws IOException {
+        String accountName = extractAccountName(uri);
+
+        if (this.openFileSystems.containsKey(accountName)) {
+            throw new FileSystemAlreadyExistsException();
```

Comment:
Should we add the already existing FileSystem name into the exception?

Reply:
Probably a good idea. I'll double check the docs, too.